vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
+    DEBUG_PRINT("Starting MC ..\n");
 
     uint32_t startTick = xTaskGetTickCount();
-    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) &&
-           ((StateInverterLeft & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED))
+
+    // Attempt to disable the lockout
+    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) && 
+           (isLockoutDisabled() == false))
     {
-        sendThrottleValueToMCs(0, 0);
+        sendLockoutReleaseToMC();
+        // mcClearFaults();
         vTaskDelay(pdMS_TO_TICKS(THROTTLE_POLL_TIME_MS));
     }
 
-    if ((StateInverterLeft & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED) {
-        ERROR_PRINT("Timeout waiting for MC Left to be ready to turn on\n");
-        ERROR_PRINT("StateInverterLeft %d\n", (uint8_t)(StateInverterLeft & INVERTER_STATE_MASK));
+
+    if ((isLockoutDisabled() == false) ||
+        (getInverterVSMState() == INV_VSM_State_FAULT_STATE)) {
+        ERROR_PRINT("Inverter lockout could not be released\n");
         return HAL_TIMEOUT;
     }
 
-    DEBUG_PRINT("Left motor controller ready.\n");
+    DEBUG_PRINT("Motor controller ready.\n");
 
-    DEBUG_PRINT("Starting MC Right...\n");
+    DEBUG_PRINT("Initializing default settings...");
 
-    startTick = xTaskGetTickCount();
-    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) &&
-           ((StateInverterRight & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED))
-    {
-        sendThrottleValueToMCs(0, 0);
-        vTaskDelay(pdMS_TO_TICKS(THROTTLE_POLL_TIME_MS));
-    }
+    // Todo - disable unused broadcasts
 
-    if ((StateInverterRight & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED) {
-        ERROR_PRINT("Timeout waiting for MC Right to be ready to turn on\n");
-        ERROR_PRINT("StateInverterRight %d\n", (uint8_t)(StateInverterRight & INVERTER_STATE_MASK));
-        return HAL_TIMEOUT;
-    }
+    // This isn't really used, but may have future use?
+    initMotorControllerSettings();
 
-    DEBUG_PRINT("Right motor controller ready.\n");
+    requestTorqueFromMC(0,0);
+    motors_active = true;
 
-    DEBUG_PRINT("Initializing default settings...");
-    initMotorControllerProcanSettings();
+    return HAL_OK;
+}
 
-    if (mcLeftCommand(INVERTER_ENABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send enable bridge command to MC Left");
-        return HAL_ERROR;
-    }
+HAL_StatusTypeDef sendDisableMC(void) {
+    VCU_INV_Inverter_Enable = INVERTER_OFF;
+    VCU_INV_Inverter_Discharge = INVERTER_DISCHARGE_DISABLE;
+    VCU_INV_Speed_Mode_Enable = SPEED_MODE_OVERRIDE_FALSE;
+    VCU_INV_Torque_Limit_Command = TORQUE_LIMIT_OVERRIDE_FALSE;
 
-    if (mcRightCommand(INVERTER_ENABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send enable bridge command to MC right");
+    if (sendCAN_MC_Command_Message() != HAL_OK) {
+        ERROR_PRINT("Failed to send disable message to MC\n");
         return HAL_ERROR;
     }
-    motors_active = true;
```

Comment:
set to false? didnt look at the details but why was this set to true. If it makes sense to you then we chilling.